scripts/executors/src/check-for-deprecated-preview-packages.ts

```diff
@@ -0,0 +1,64 @@
+import { execSync } from 'child_process';
+import { readFileSync, readdirSync } from 'fs';
+import { join } from 'path';
+
+import { logger } from 'just-scripts';
+import yargs from 'yargs';
+
+/**
+ * Deprecates a package by executing an `npm deprecate` command.
+ *
+ * @param name - The name of the package to deprecate.
+ * @param npmToken - The npm authentication token.
+ * @returns A promise that resolves when the deprecation is successful, or rejects with an error if it fails.
+ */
+export function deprecatePackage(name: string, npmToken: string) {
+  const command = `npm deprecate ${name} "Deprecated in favor of stable release" --registry https://registry.npmjs.org/ --//registry.npmjs.org/:_authToken=${npmToken}`;
+
+  logger.info(command);
+
+  try {
+    execSync(command, { stdio: 'inherit' });
+  } catch (e) {
+    logger.error(`Failed to deprecate "${name}" package`);
+  }
+}
+
+/**
+ * Retrieves the list of packages to deprecate based on change files in the `change` directory.
+ * A package is marked for deprecation if:
+ *   - The change file has a type of `minor`
+ *   - The change file has a message of `chore: release stable`
+ *
+ * @returns {string[]} An array of package names to deprecate.
+ */
+export function getPackagesToDeprecate() {
+  const changeDir = join(__dirname, 'change');
+
+  const packagesToDeprecate = readdirSync(changeDir).reduce<string[]>((acc, file) => {
+    const filePath = join(changeDir, file);
+    const fileContent = readFileSync(filePath, 'utf-8');
+    const changeFile = JSON.parse(fileContent);
+
+    if (changeFile.type === 'minor' && changeFile.message === 'chore: release stable') {
```

Comment:
this isn't most robust assumption about change-file ( we don't have any validators(yet) on PRs that contributor actually used generated changefile when going from preview to stable ).

because of that we need to gather more data in order to identify preview -> stable release 
- contains tag `vNext"
- name doesn't end with `-preview`
- version is `9.0.0-alpha.0` / https://github.com/microsoft/fluentui/blob/master/tools/workspace-plugin/src/generators/prepare-initial-release/index.spec.ts#L408-L412
- change type is `minor` / https://github.com/microsoft/fluentui/blob/master/tools/workspace-plugin/src/generators/prepare-initial-release/index.spec.ts#L525-L527
-  change file `@fluentui-<package-name>-hash.json`

Reply:
> this isn't most robust assumption about change-file ( we don't have any validators(yet) on PRs that contributor actually used generated changefile when going from preview to stable ).

I followed the logic outlined in the issue 😄. Nonetheless, I have now updated it to the method you suggested in the comment.

Reply:
yeah issue content might get partially obsolete/incomplete in time as requirements change 🥹, ty